core/remotes/resolver.go

```diff
@@ -43,7 +47,7 @@ type Resolver interface {
 	// Fetcher returns a new fetcher for the provided reference.
 	// All content fetched from the returned fetcher will be
 	// from the namespace referred to by ref.
-	Fetcher(ctx context.Context, ref string) (Fetcher, error)
+	Fetcher(ctx context.Context, ref string, opts ...FetcherOpt) (Fetcher, error)
```

Comment:
This will be a breaking change to this interface.

Reply:
Yes, hopefully not too bad given the var-args, but yes. But yes, I see what you mean, the "if you say yes, it's forever" kid of breaking change.